app/helpers/samples_helper.rb

```diff
@@ -513,7 +513,14 @@ def upload_samples_with_metadata(samples_to_upload, metadata, user)
 
       if sample_attributes[:host_genome_name]
         name = sample_attributes.delete(:host_genome_name)
-        sample_attributes[:host_genome_id] = HostGenome.find_by(name: name).id
+        begin
+          hg = HostGenome.find_or_create_by!(name: name)
```

Comment:
Shouldnt this be guarded by feature flag?

Reply:
This functionality will be hidden from users until there is the new input is in place, which I was planning to put behind a feature flag.

Reply:
I see, so, it assumes that users will not have any choice other than existing host genomes.

ok, up to you, but, imho, seems to me this should not be an isolated change, since a lot more goes into creating host genomes... its a bit risky. 

Is there a design doc and/or product spec to have a better idea of this feature?

Reply:
The idea is to enable users to create new host genomes at upload time. The new host genomes will default to using ERCC index files, effectively doing what we tell users to do now -- choose the "ercc only" option if their host is not in the list. 

I'd be happy to put this in admin only if you think that's best.

Reply:
I see. So, it will be ok to create host genomes with name only?

I do not have the full picture of the sequence of tasks and timeline to make the whole feature work, so I will leave it up to you to evaluate the risk - if any - of leaving this open.

Reply:
I added the `admin` gating 👍